tools/bcdb_mock/pkg/workloads/types/reservation.go

```diff
@@ -43,6 +45,28 @@ const (
 	Deleted = generated.NextActionDeleted
 )
 
+// ReservationQuery type to parse reservation query string
+type ReservationQuery struct {
```

Comment:
It's confusing that we have a reservation query and a filter. Why this can't be 

`filter.WithCustomerID` and filter.WithNextAction` for the parse method this can be also implemented on the filter itself! no ?